engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -43,19 +44,27 @@
     EntityManager entityManager;
 
     @ReceiveEvent
-    public void onAutoMountCamera(OnChangedComponent event, EntityRef client, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
-        if (clientComponent.camera.exists()) {
-            client.send(new ResetCameraEvent());
+    public void ensureCameraEntityCreatedOnChangedClientComponent(OnChangedComponent event, EntityRef client, AutoMountCameraComponent autoMountCameraComponent, ClientComponent clientComponent) {
+        if (localPlayer.getClientEntity().equals(client)) {
+            ensureCameraEntityCreated();
         }
     }
 
```

Comment:
I like your very verbose method names. Very clear! They are almost entirely self-documenting! 

I wonder if some javadocs for class and public methods would still be useful though. Perhaps offering some context to what's going on here?